app.js

```diff
@@ -16,41 +27,92 @@ app.use(require('body-parser').json());
 
 app.get('/reviews/:reviewId', function(req, res) {
-  
-  var _arrayOfReviews = [];
-
-  //number of pages
-  cloudscraper.get(review_url + req.params.reviewId + review_page + 1, function(error, response, body) {
-    if (error) {
-        return console.error(error);
-    } else {
-      var pageList = [];
-      var $ = cheerio.load(body);
-      $("li[class='page-button']").each(function(i, element){
-          pageList.push(parseInt($(this).text()));
-      });
-      var numberOfPages = pageList.pop();
-
-      scrapeEveryPage(numberOfPages, _arrayOfReviews, req.params.reviewId).then(function (options) {
-          res.send(options);
-      });
+  var reviewId = req.params.reviewId;
+  existingCloudantDoc(reviewId).then(function (result) {
+    if (result) {
+      console.log(reviewId + " document exists");
+      scrapeNumberOfReviews(reviewId)
+        .then(isNumberOfReviewsEqual)
+        .then(function (options) {
+          if (options.isEqual) {
+            res.send("Query Watson Discovery if Collection for product exists.")
+            // remove res.send and query if discovery collection exists
+          }
+          else {
+            scrapeNumberOfPages(reviewId)
+              .then(scrapeEveryPage)
+              .then(function (_options) {
+                var cloudantDocument = {
+                  _id: _options.productId,
+                  productName: _options.productName,
+                  reviews: _options.reviews,
+                  _rev: options._rev
+                };
+                insertCloudantDoc(cloudantDocument)
+                  .then(function (options) {
+                    res.send("Updated document in Cloudant");
+                    // remove res.send and query watson discovery if collection exists
+                  });
+              })
+          }
+        });
+    }
+    else {
+      console.log(reviewId + " document doesnt exist");
+      scrapeNumberOfPages(reviewId)
+        .then(scrapeEveryPage)
+        .then(function (options) {
+          var cloudantDocument = {
+            _id: options.productId,
+            productName: options.productName,
+            reviews: options.reviews
+          };
+          insertCloudantDoc(cloudantDocument)
+            .then(function (options) {
+              res.send("Saved document in Cloudant");
+              // remove res.send and create collecion in watson discovery
+            })
+        });
     }
   });
 });
 
+app.get('/cloudant/:reviewId', function(req, res) {
+  getCloudantReviews(req.params.reviewId)
+    .then(function (result) {
+      res.send(result);
+    });
+});
+
 var port = process.env.PORT || 4000
 app.listen(port, function() {
   console.log("Oder API service is in port: " + port);
 });
 
-function scrapeEveryPage(totalPages, arrayOfReviews, productId) {
+/**
+ * Scrape every page for the reviews
+ * @param {object} options - product ID of Amazon Product found in URL
+ * options = {
+ * "totalPages": 30,
```

Comment:
how is the number of pages passed in?

Reply:
`scrapeNumberOfPages` resolves an object
```
object = {
 "productId" = "B0123",
 "totalPages" : 30
}
```
The object is then used here

Reply:
ah gotcha.